common/utils/src/main/scala/org/apache/spark/SparkThrowableHelper.scala

```diff
@@ -51,9 +51,11 @@ private[spark] object SparkThrowableHelper {
       messageParameters: Map[String, String],
       context: String): String = {
     val displayMessage = errorReader.getErrorMessage(errorClass, messageParameters)
+    val sqlState = errorReader.getSqlState(errorClass)
+    val displaySqlState = if (sqlState == null) "" else s" SQLSTATE: $sqlState"
```

Comment:
```suggestion
    val displaySqlState = if (sqlState == null) "undefined" else s" SQLSTATE: $sqlState"
```

Reply:
Not a fan.... I'd have to say: SQLSTATE: undefined and confuse any tools.
Note that this only happens for LEGACY. Not posting it as we do today seems better to me.

Reply:
I agree. `undefined` is useless for users. We should catch them at our site during testing.